src-ui/src/app/components/manage/correspondent-list/correspondent-list.component.html

```diff
@@ -52,3 +52,8 @@
     </tr>
     </tbody>
   </table>
+
```

Comment:
seems to me that this could be an own component to avoid duplication. `<app-paginator>` or something like this.

Reply:
I just spent a little time doing this and tbh in the end it I wonder if its unnecessary. We already use `ngb-pagination` so the duplication is mostly just a little html markup around that. Extracting to a new component turns out to be like 10x more new lines than the duplicated ones it avoids =)

I defer to you on this, attached is a patch that implements it completely, LMK what you think. Im good either way.

```
From faf30f127511f8b14d2768a4e40d787ae8e7dd1c Mon Sep 17 00:00:00 2001
From: Michael Shamoon <4887959+shamoon@users.noreply.github.com>
Date: Sat, 19 Mar 2022 00:20:20 -0700
Subject: [PATCH] Extract management pagination to component

---
 src-ui/src/app/app.module.ts                  |  2 ++
 .../correspondent-list.component.html         | 29 +++++++++----------
 .../document-type-list.component.html         |  5 +---
 .../management-list-pagination.component.html |  4 +++
 .../management-list-pagination.component.scss |  0
 ...nagement-list-pagination.component.spec.ts | 24 +++++++++++++++
 .../management-list-pagination.component.ts   | 19 ++++++++++++
 .../manage/tag-list/tag-list.component.html   |  5 +---
 8 files changed, 64 insertions(+), 24 deletions(-)
 create mode 100644 src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.html
 create mode 100644 src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.scss
 create mode 100644 src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.spec.ts
 create mode 100644 src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.ts

diff --git a/src-ui/src/app/app.module.ts b/src-ui/src/app/app.module.ts
index 772d40ae..cf9dccf1 100644
--- a/src-ui/src/app/app.module.ts
+++ b/src-ui/src/app/app.module.ts
@@ -67,6 +67,7 @@ import { ApiVersionInterceptor } from './interceptors/api-version.interceptor'
 import { ColorSliderModule } from 'ngx-color/slider'
 import { ColorComponent } from './components/common/input/color/color.component'
 import { DocumentAsnComponent } from './components/document-asn/document-asn.component'
+import { ManagementListPaginationComponent } from './components/manage/management-list-pagination/management-list-pagination.component'
 
 import localeCs from '@angular/common/locales/cs'
 import localeDa from '@angular/common/locales/da'
@@ -153,6 +154,7 @@ registerLocaleData(localeZh)
     DateComponent,
     ColorComponent,
     DocumentAsnComponent,
+    ManagementListPaginationComponent,
   ],
   imports: [
     BrowserModule,
diff --git a/src-ui/src/app/components/manage/correspondent-list/correspondent-list.component.html b/src-ui/src/app/components/manage/correspondent-list/correspondent-list.component.html
index 2b033058..79df4b98 100644
--- a/src-ui/src/app/components/manage/correspondent-list/correspondent-list.component.html
+++ b/src-ui/src/app/components/manage/correspondent-list/correspondent-list.component.html
@@ -14,16 +14,16 @@
 </div>
 
 <table class="table table-striped align-middle border shadow-sm">
-    <thead>
-    <tr>
-      <th scope="col" sortable="name" [currentSortField]="sortField" [currentSortReverse]="sortReverse" (sort)="onSort($event)" i18n>Name</th>
-      <th scope="col" sortable="matching_algorithm" [currentSortField]="sortField" [currentSortReverse]="sortReverse" (sort)="onSort($event)" i18n>Matching</th>
-      <th scope="col" sortable="document_count" [currentSortField]="sortField" [currentSortReverse]="sortReverse" (sort)="onSort($event)" i18n>Document count</th>
-      <th scope="col" sortable="last_correspondence" [currentSortField]="sortField" [currentSortReverse]="sortReverse" (sort)="onSort($event)" i18n>Last correspondence</th>
-      <th scope="col" i18n>Actions</th>
-    </tr>
-    </thead>
-    <tbody>
+  <thead>
+  <tr>
+    <th scope="col" sortable="name" [currentSortField]="sortField" [currentSortReverse]="sortReverse" (sort)="onSort($event)" i18n>Name</th>
+    <th scope="col" sortable="matching_algorithm" [currentSortField]="sortField" [currentSortReverse]="sortReverse" (sort)="onSort($event)" i18n>Matching</th>
+    <th scope="col" sortable="document_count" [currentSortField]="sortField" [currentSortReverse]="sortReverse" (sort)="onSort($event)" i18n>Document count</th>
+    <th scope="col" sortable="last_correspondence" [currentSortField]="sortField" [currentSortReverse]="sortReverse" (sort)="onSort($event)" i18n>Last correspondence</th>
+    <th scope="col" i18n>Actions</th>
+  </tr>
+  </thead>
+  <tbody>
     <tr *ngFor="let correspondent of data">
       <td scope="row">{{ correspondent.name }}</td>
       <td scope="row">{{ getMatching(correspondent) }}</td>
@@ -50,10 +50,7 @@
           </div>
         </td>
     </tr>
-    </tbody>
-  </table>
+  </tbody>
+</table>
 
-  <div class="d-flex">
-    <div i18n *ngIf="collectionSize > 0">{collectionSize, plural, =1 {One correspondent} other {{{collectionSize || 0}} total correspondents}}</div>
-    <ngb-pagination *ngIf="collectionSize > 20" class="ms-auto" [pageSize]="25" [collectionSize]="collectionSize" [(page)]="page" (pageChange)="reloadData()" aria-label="Default pagination"></ngb-pagination>
-  </div>
+<app-management-list-pagination [collectionSize]="collectionSize" (pageChange)="reloadData()" [(page)]="page" itemName="correspondent"></app-management-list-pagination>
diff --git a/src-ui/src/app/components/manage/document-type-list/document-type-list.component.html b/src-ui/src/app/components/manage/document-type-list/document-type-list.component.html
index 32a47dcd..ccd49f81 100644
--- a/src-ui/src/app/components/manage/document-type-list/document-type-list.component.html
+++ b/src-ui/src/app/components/manage/document-type-list/document-type-list.component.html
@@ -51,7 +51,4 @@
   </tbody>
 </table>
 
-<div class="d-flex">
-  <div i18n *ngIf="collectionSize > 0">{collectionSize, plural, =1 {One document type} other {{{collectionSize || 0}} total document types}}</div>
-  <ngb-pagination *ngIf="collectionSize > 20" class="ms-auto" [pageSize]="25" [collectionSize]="collectionSize" [(page)]="page" (pageChange)="reloadData()" aria-label="Default pagination"></ngb-pagination>
-</div>
+<app-management-list-pagination [collectionSize]="collectionSize" (pageChange)="reloadData()" [(page)]="page" itemName="document type"></app-management-list-pagination>
diff --git a/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.html b/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.html
new file mode 100644
index 00000000..fcfca958
--- /dev/null
+++ b/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.html
@@ -0,0 +1,4 @@
+<div class="d-flex">
+  <div i18n *ngIf="collectionSize > 0">{collectionSize, plural, =1 {One {{itemName}}} other {{{collectionSize || 0}} total {{itemName}}s}}</div>
+  <ngb-pagination *ngIf="collectionSize > 20" class="ms-auto" [pageSize]="25" [collectionSize]="collectionSize" [(page)]="page" (pageChange)="pageChange.emit()" aria-label="Default pagination"></ngb-pagination>
+</div>
diff --git a/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.scss b/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.scss
new file mode 100644
index 00000000..e69de29b
diff --git a/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.spec.ts b/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.spec.ts
new file mode 100644
index 00000000..9b8527fc
--- /dev/null
+++ b/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.spec.ts
@@ -0,0 +1,24 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+
+import { ManagementListPaginationComponent } from './management-list-pagination'
+
+describe('ManagementListPaginationComponent', () => {
+  let component: ManagementListPaginationComponent
+  let fixture: ComponentFixture<ManagementListPaginationComponent>
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ManagementListPaginationComponent],
+    }).compileComponents()
+  })
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ManagementListPaginationComponent)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+  })
+
+  it('should create', () => {
+    expect(component).toBeTruthy()
+  })
+})
diff --git a/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.ts b/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.ts
new file mode 100644
index 00000000..f111fc4a
--- /dev/null
+++ b/src-ui/src/app/components/manage/management-list-pagination/management-list-pagination.component.ts
@@ -0,0 +1,19 @@
+import { Component, EventEmitter, Input } from '@angular/core'
+
+@Component({
+  selector: 'app-management-list-pagination',
+  templateUrl: './management-list-pagination.component.html',
+  styleUrls: ['./management-list-pagination.component.scss'],
+})
+export class ManagementListPaginationComponent {
+  @Input()
+  collectionSize: number
+
+  @Input()
+  itemName: string
+
+  @Input()
+  page: number
+
+  pageChange: EventEmitter<number> = new EventEmitter()
+}
diff --git a/src-ui/src/app/components/manage/tag-list/tag-list.component.html b/src-ui/src/app/components/manage/tag-list/tag-list.component.html
index d51ad7d3..2909f62b 100644
--- a/src-ui/src/app/components/manage/tag-list/tag-list.component.html
+++ b/src-ui/src/app/components/manage/tag-list/tag-list.component.html
@@ -54,7 +54,4 @@
   </tbody>
 </table>
 
-<div class="d-flex">
-  <div i18n *ngIf="collectionSize > 0">{collectionSize, plural, =1 {One tag} other {{{collectionSize || 0}} total tags}}</div>
-  <ngb-pagination *ngIf="collectionSize > 20" class="ms-auto" [pageSize]="25" [collectionSize]="collectionSize" [(page)]="page" (pageChange)="reloadData()" aria-label="Default pagination"></ngb-pagination>
-</div>
+<app-management-list-pagination [collectionSize]="collectionSize" (pageChange)="reloadData()" [(page)]="page" itemName="tag"></app-management-list-pagination>
-- 
2.30.1
```

Reply:
Sure, adding a new component is always some boilerplate code. For me, it is not the duplication of code lines. it is the duplication of (translation-) logic. If you remove the spec file, which is useless in this case, because it tests nothing, your patch will shrink.

But for now, I am okay with that. I want that we keep this in mind if we maybe add more listviews to the frontend we can have a second look.

Reply:
I have an idea that will supersede this, basically I have reworked (still in progress) the management list to use a single unified component, are you OK merging this for now and then we can discuss that idea separately?

Reply:
sure we can make a follow-up from this. Thanks.